src/components/Input/Input.tsx

```diff
@@ -1,54 +1,60 @@
-import * as PropTypes from 'prop-types'
 import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as cx from 'classnames'
 import * as _ from 'lodash'
 
 import {
   AutoControlledComponent,
-  createHTMLInput,
   customPropTypes,
-  getUnhandledProps,
+  IRenderResultConfig,
   partitionHTMLProps,
 } from '../../lib'
-import Icon from '../Icon'
-import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
 import {
-  ComponentEventHandler,
   Extendable,
-  ReactChildren,
-  ShorthandRenderFunction,
   ShorthandValue,
+  ShorthandRenderFunction,
+  ComponentEventHandler,
 } from '../../../types/utils'
+import { ComponentPartStyle, ComponentVariablesInput } from 'theme'
+import Icon from '../Icon'
+import Slot from '../Slot'
+import Ref from '../Ref'
 
 export interface IInputProps {
   as?: any
-  children?: ReactChildren
   className?: string
   clearable?: boolean
-  defaultValue?: string | number
+  defaultValue?: React.ReactText
```

Comment:
I would vote for leaving it as it was before, because of its explicitness

Reply:
we'd duplicate this `string | number` type everywhere in the file (3 places); I vote for consistency over explicitness if u are not really against it